client/signup/steps/store-features/index.tsx

```diff
@@ -21,6 +24,7 @@ interface Props {
 	signupDependencies: any;
 	stepName: string;
 	initialContext: any;
+	siteId: number;
```

Comment:
I believe `siteId` is a number type, but `getSiteEditorUrl` expects it to be an object, so there's a type mismatch. I wonder if the typing for `getSiteEditorUrl` is a typo? I don't think I've ever seen a site ID passed as an object.

Reply:
Looking through Calypso I see most of the time site ID is typed as a number.

Reply:
I traced a bunch of it, I can't see anything that uses the site ID as an object. 🤔 @Addison-Stavlo do you know why `siteID` is [expected to be an object](https://github.com/Automattic/wp-calypso/blob/7655b6d1f326371990851da10ebb97f133fbdc17/client/state/selectors/get-site-editor-url.js#L8) by `getSiteEditorUrl`?

Reply:
Couple paths forward to un-block this:

* Update the expected param type in `getSiteEditorUrl` for `siteId` to `number` (this seems like the logical type for this value) or `number|object` (latter may be safer in the event some other code is also passing it as an object)
* Stop using `getSiteEditorUrl` in favor of a hard-coded path `/site-editor/${siteSlug}/` as a temp workaround; we bypass some logic around `shouldLoadGutenframe` but I thought Gutenframe was dead, so....maybe not a big deal.

Reply:
Let's try `number|object` (I wouldn't want to walk around Gutenframe related stuff, we still use it all the time for the Calypso editor).

Reply:
> I believe `siteId` is a number type, but `getSiteEditorUrl` expects it to be an object

This has to be an error in the `getSiteEditorUrl` doc comment, all the functions it ends up calling (`getSiteAdminUrl`, `shouldLoadGutenframe`, `getSiteSlug`, etc) expect a number for siteId ... I think you're good to update the doc comment!

Reply:
Awesome, I've updated `getSiteEditorUrl` to use the correct type!